src/ui/setup/index.ts

```diff
@@ -59,8 +67,10 @@ export function bootstrapApp() {
       setTelemetryContext(userInfo);
     }
 
-    const userSettings = await getUserSettings();
-    store.dispatch(setWorkspaceId(userSettings.defaultWorkspaceId));
+    if (!getWorkspaceId(store.getState())) {
```

Comment:
Is this just so we don't bother querying+updating the workspaceId if they're going through the billing path?

Reply:
No, it's because in this case we already set the `workspaceId` (to the one from the route) and don't want to change that to the user's default `workspaceId`.